coffee/base.py

```diff
@@ -466,6 +466,14 @@ def is_const(self):
         return not self.rank or all(is_const_dim(r) for r in self.rank)
 
     @property
+    def is_number(self):
```

Comment:
I have a list of changes for COFFEE in mind including a restructuring of ``base.py``, as well as the addition of new types (eg ``Number``) and class attributes (eg ``is_Symbol``, ``is_Incr``). I'll do this in another PR